ledger/sandbox/src/main/scala/com/digitalasset/platform/store/ActiveLedgerStateManager.scala

```diff
@@ -224,8 +226,22 @@ class ActiveLedgerStateManager[ALS <: ActiveLedgerState[ALS]](initialState: => A
     val divulgedContractIds = globalDivulgence -- st.archivedIds
     st.mapAcs(
         _ divulgeAlreadyCommittedContracts (transactionId, divulgedContractIds, divulgedContracts))
-      .mapAcs(_ addParties st.parties)
+      .mapAcs { acs =>
+        if (indexingOptions.implicitPartyAllocation)
+          acs addParties st.parties
+        else
+          acs
+      }
       .result
   }
 
 }
+
+object ActiveLedgerStateManager {
+
+  final case class IndexingOptions(implicitPartyAllocation: Boolean)
+
+  object IndexingOptions {
+    val defaultNoImplicitPartyAllocation = IndexingOptions(implicitPartyAllocation = false)
```

Comment:
Perhaps rename this to just `default`? If you need to know the contents, you don't want the default.